src/components/Demonstration/TreeNetworkGraph.jsx

```diff
@@ -54,36 +51,112 @@ function TreeNetworkGraph({ data, width, height, onNodeClick = () => {} }) {
           })
           .strength(0.8)
       )
+  ).current
+  const depth = Math.max(...data.nodes.map(e => e.level))
+  const executionId = useMemo(() => data?.nodes[0]?.executionId, [data])
```

Comment:
Any reason you use a `useMemo` for executionId and not for depth?

Reply:
It solved the problem of constant refresh so I stopped there but I can change that